db/view/view.cc

```diff
@@ -1364,7 +1368,13 @@ future<std::optional<utils::chunked_vector<frozen_mutation_and_schema>>> view_up
         co_await advance_existings();
     }
 
-    while (co_await on_results() == stop_iteration::no) {};
+    // If the partition tombstone update is applied to all the views and there are no other updates, we can skip going over
+    // all the rows trying to generate row updates, because the partition tombstones already cover everything.
+    if (is_partition_tombstone_applied_on_all_views &&  _update->is_end_of_partition()) {
+        _skip_row_updates = true;
```

Comment:
I didn't understand why this _skip_row_updates is a class variable and not a local variable in this specific call. Do we need to remember it later too?

Reply:
yes, we need to remember it for the next call. Maybe I should arrange the code differently to avoid it.
this function `build_some` is called until it doesn't generate any new updates. 
So in the first call it generates the partition tombstone update, and then it is called again and we need to remember to skip everything, and we return with no new updates and break the loop.
Alternatively we could maybe return an indication to stop the iteration.